connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConnector.java

```diff
@@ -0,0 +1,140 @@
+                           ConnectorContext ctx,
+                           ConnectorStatus.Listener statusListener) {
+        this.connName = connName;
+        this.ctx = ctx;
+        this.connector = connector;
+        this.statusListener = statusListener;
+        this.state = null;
+    }
+
+    public synchronized void initialize(Map<String, String> config) {
+        log.debug("Initializing connector {} with config {}", connName, config);
+
+        this.config = config;
+
+        connector.initialize(new ConnectorContext() {
+            @Override
+            public void requestTaskReconfiguration() {
+                ctx.requestTaskReconfiguration();
+            }
+
+            @Override
+            public void raiseError(Exception e) {
+                WorkerConnector.this.state = State.FAILED;
+                ctx.raiseError(e);
+            }
+        });
+    }
+
+    private void start() {
+        try {
+            if (state == State.STARTED)
+                return;
+            connector.start(config);
+            statusListener.onStartup(connName);
+            this.state = State.STARTED;
+        } catch (Throwable t) {
+            log.error("Error while starting connector {}", connName, t);
+            statusListener.onFailure(connName, t);
+            this.state = State.FAILED;
+        }
+    }
+
+    private void pause() {
+        try {
+            if (state == State.STOPPED)
+                return;
+
+            if (state == State.STARTED)
+                connector.stop();
+
+            statusListener.onPause(connName);
+            this.state = State.STOPPED;
+        } catch (Throwable t) {
+            log.error("Error while shutting down connector {}", connName, t);
+            statusListener.onFailure(connName, t);
+            this.state = State.FAILED;
+        }
+    }
+
+    public synchronized void shutdown() {
+        try {
+            if (state == State.STARTED)
+                connector.stop();
+            this.state = State.STOPPED;
+        } catch (Throwable t) {
+            log.error("Error while shutting down connector {}", connName, t);
+            this.state = State.FAILED;
+        } finally {
+            statusListener.onShutdown(connName);
+        }
+    }
+
+    public synchronized void transitionTo(TargetState targetState) {
+        log.debug("Transition connector {} to {}", connName, targetState);
+        if (targetState == TargetState.PAUSED) {
+            pause();
+        } else if (targetState == TargetState.STARTED) {
+            start();
+        } else {
+            throw new IllegalArgumentException("Unhandled target state " + targetState);
+        }
+    }
+
+    public boolean isSinkConnector() {
+        return SinkConnector.class.isAssignableFrom(connector.getClass());
+    }
+
+    public Connector connector() {
+        return connector;
+    }
+
+    @Override
+    public String toString() {
+        return connector.toString();
```

Comment:
I realize this is carried over and the code simply moved, but I'm noticing now that it can be misleading. Do we actually want the string representation to be the same for both the connector and this class? I'm not sure we ever convert to a string, but if it shows up in logging or something it seems like it'd be confusing.